mentoring/components/questionnaire.py

```diff
@@ -56,34 +51,36 @@ class QuestionnaireAbstractBlock(LightChild, StepMixin):
                    default=1, scope=Scope.content, enforce_type=True)
 
     valid_types = ('choices')
+    has_children = True
 
     @classmethod
-    def init_block_from_node(cls, block, node, attr):
-        block.light_children = []
-        for child_id, xml_child in enumerate(node):
+    def parse_xml(cls, node, runtime, keys, id_generator):
+        block = runtime.construct_xblock_from_class(cls, keys)
+
+        # Load XBlock properties from the XML attributes:
+        for name, value in node.items():
+            setattr(block, name, value)
+
+        for xml_child in node:
             if xml_child.tag == 'question':
                 block.question = xml_child.text
             elif xml_child.tag == 'message' and xml_child.get('type') == 'on-submit':
                 block.message = (xml_child.text or '').strip()
-            else:
-                cls.add_node_as_child(block, xml_child, child_id)
-
-        for name, value in attr:
-            setattr(block, name, value)
+            elif xml_child.tag is not etree.Comment:
```

Comment:
@bradenmacdonald mostly because when I did something similar I ended up going with comment-ignoring parser. I'm sorry for using "incorrect" (not actually using, but it sounded like that), I meant more like "this would work, but there's potentially better way". Please forgive me, I'm not native english-speaker :)